### DISCUSSION TITLE
User system add the quota limit method

### DISCUSSION THREAD

**Discussion by Zeta-ZhaoLi** [open-webui/open-webui#3607] [General]
**Is your feature request related to a problem? Please describe.**

No.

**Describe the solution you'd like**

Every user should be able to set their own API Key, use their own quota.
Or even further more, add the quota data, and add a price data for every model (charge for tokens or requests), then will be able to limit the usage for every user. So that the admin can only set the main Key.

**Describe alternatives you've considered**

Every user use their own Key should be easy to modify the system.

**Additional context**

We all know the long chat costs quota a lot, there should be a way to limit every user, or even just know everyone's usage.
The road map mentioned a finely control access level, but no specific describe, if my issue already in your plan, just ignore this.

**Comment by tjbck** [comment]
Related: https://openwebui.com/f/hub/openai_manifold/

**Reply by Zeta-ZhaoLi** [reply]
@tjbck Thank you for your link, but I'm having a problem with this function:

I have installed and set the fucntion, it got the model list succesfully, but I can't set the value for the users, as the screenshot shows.

Do you have any idea about this issue?

open-webui version: v0.3.7

![image](https://github.com/open-webui/open-webui/assets/9345171/2c83e54b-2390-4553-82cc-be90a25eb1bd)

**Reply by Zeta-ZhaoLi** [reply]
I found that's because of the new version, I rolled back to v0.3.6 then works.

And then part of the users works, but the others can't see the setting of OpenAI Manifold value still.

By the way, it seems the setting of values are unable to save for a long time, I always need to fill it if I switched my account.

**Reply by justinh-rahb** [reply]
This issue is fixed in 0.3.8 dev.

**Reply by Zeta-ZhaoLi** [reply]
> This issue is fixed in 0.3.8 dev.

0.3.8 dev works well, thank you for your answer.
